utxonursery_test.go

```diff
@@ -0,0 +1,332 @@
+			t.Fatalf("unable to serialize and add kid output to list: %v", err)
+		}
+	}
+
+	kidList, err := deserializeKidList(&b)
+	if err != nil {
+		t.Fatalf("unable to deserialize kid output list: ", err)
+	}
+
+	for i := 0; i < 3; i++ {
+		if kidOutputs[i].amt != kidList[i].amt {
+			t.Fatalf("amounts don't match: %v vs. %v", kidOutputs[i].amt, kidList[i].amt)
+		}
+
+		if kidOutputs[i].outPoint != kidList[i].outPoint {
+			t.Fatalf("outputs don't match: %v vs. %v", kidOutputs[i].outPoint, kidList[i].outPoint)
+		}
+
+		if kidOutputs[i].blocksToMaturity != kidList[i].blocksToMaturity {
+			t.Fatalf("blocks to maturity doesn't match: %v vs. %v", kidOutputs[i].blocksToMaturity,
+				kidList[i].blocksToMaturity)
+		}
+
+		if kidOutputs[i].confHeight != kidList[i].confHeight {
+			t.Fatalf("confirmation height doesn't match: %v vs. %v", kidOutputs[i].confHeight,
+				kidList[i].confHeight)
+		}
+		if !kidOutputs[i].signDescriptor.PubKey.IsEqual(kidList[i].signDescriptor.PubKey) {
+			t.Fatalf("pubkeys don't match: %v vs. %v", kidOutputs[i].signDescriptor.PubKey,
+				kidList[i].signDescriptor.PubKey)
+		}
+
+		if !bytes.Equal(kidOutputs[i].signDescriptor.PrivateTweak, kidList[i].signDescriptor.PrivateTweak) {
+			t.Fatalf("private tweaks don't match: %v vs. %v",
+				kidOutputs[i].signDescriptor.PrivateTweak, kidList[i].signDescriptor.PrivateTweak)
+		}
+
+		if !bytes.Equal(kidOutputs[i].signDescriptor.WitnessScript, kidList[i].signDescriptor.WitnessScript) {
+			t.Fatalf("witness scripts don't match: %v vs. %v",
+				kidOutputs[i].signDescriptor.WitnessScript, kidList[i].signDescriptor.WitnessScript)
+		}
+
+		if kidOutputs[i].signDescriptor.Output.Value != kidList[i].signDescriptor.Output.Value {
+			t.Fatalf("output values don't match: %v vs. %v", kidOutputs[i].signDescriptor.Output.Value,
+				kidList[i].signDescriptor.Output.Value)
+		}
+
+		if !bytes.Equal(kidOutputs[i].signDescriptor.Output.PkScript, kidList[i].signDescriptor.Output.PkScript) {
+			t.Fatalf("output scripts don't match: %v vs. %v", kidOutputs[i].signDescriptor.Output.PkScript,
+				kidList[i].signDescriptor.Output.PkScript)
+		}
+
+		if kidOutputs[i].signDescriptor.HashType != kidList[i].signDescriptor.HashType {
+			t.Fatalf("hashtypes don't match: %v vs. %v", kidOutputs[i].signDescriptor.HashType,
+				kidList[i].signDescriptor.HashType)
+		}
+	}
+}
+
+func TestSerializeKidOutput(t *testing.T) {
+	kid := &kidOutputs[0]
+	descriptor := &signDescriptors[0]
+	pk, err := btcec.ParsePubKey(keys[0], btcec.S256())
+	if err != nil {
+		t.Fatalf("unable to parse pub key: %v", keys[0])
+	}
+	descriptor.PubKey = pk
+	kid.signDescriptor = descriptor
+
+	var b bytes.Buffer
+
+	if err := serializeKidOutput(&b, kid); err != nil {
+		t.Fatalf("unable to serialize kid output: %v", err)
+	}
+
+	deserializedKid, err := deserializeKidOutput(&b)
+	if err != nil {
+		fmt.Printf(err.Error())
+	}
+
+	if kid.amt != deserializedKid.amt {
```

Comment:
A call to a single call to `relect.DeepEqual(kid, deserializedKid)` may be able to replace much of the comparison code below here.